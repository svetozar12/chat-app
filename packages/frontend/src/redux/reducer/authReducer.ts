import { Action } from "../state";
import { ActionType } from "../state";
import { InitialState } from "../state";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  remember_me: false,
  input_username: "",
  input_password: "",
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: "",
};
const reducer = (state: InitialState = initialState, action: Action) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.SAVE_INPUT_USERNAME:
      return { ...state, input_username: action.payload };
    case ActionType.SAVE_INPUT_PASSWORD:
      return { ...state, input_password: action.payload };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    case ActionType.SIGN_OUT:
      return { state: undefined };
    default:
      return state;
  }
};

export default reducer;
export { initialState as AuthState };
