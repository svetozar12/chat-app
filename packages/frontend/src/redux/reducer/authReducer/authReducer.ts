import { HYDRATE } from "next-redux-wrapper";
import { IAuthState } from "./state";
import { ActionType } from "../../types";
export const initialState: IAuthState = {
  remember_me: false,
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: {
    username: "",
    id: "",
    token: "",
    refresh_token: "",
  },
};
const reducer = (state = initialState, action: any) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default reducer;
export { initialState as AuthState };
