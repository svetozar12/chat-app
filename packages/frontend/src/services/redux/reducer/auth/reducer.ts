import { HYDRATE } from 'next-redux-wrapper';
import { IAuth } from './state';
import { ActionType } from '../../types';

export const initialState: IAuth = {
  remember_me: false,
  loginPrompt: false,
};
const authReducer = (state = initialState, action: any) => {
  switch (action.type) {
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default authReducer;
