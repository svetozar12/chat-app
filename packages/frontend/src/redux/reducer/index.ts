import { combineReducers } from "redux";
import reducer from "./authReducer";
import setReducer from "./setReducer";
import saveInputReducer from "./save_inputReducer";
import messageReducer from "./messageReducer";

const reducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

export default reducers;
