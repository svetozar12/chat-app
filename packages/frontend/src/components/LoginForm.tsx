import { useSelector, useDispatch } from "react-redux";
import Link from "next/dist/client/link";
import { InitialState } from "../redux/state";

function LoginForm({
  handleSubmit,
}: {
  handleSubmit: (e: React.MouseEvent<HTMLButtonElement>) => Promise<void>;
}) {
  const state = useSelector(
    (state: { authReducer: InitialState }) => state.authReducer,
  );
  const dispatch = useDispatch();
  return (
    <div className="login-box">
      <form style={{ height: "100vh" }} className="container">
        <h1 className="alert" style={{ color: "red" }}>
          {state.bad}
        </h1>
        <h1>Login</h1>
        <input
          value={state.input_username}
          onChange={(e) =>
            dispatch({ type: "SAVE_INPUT_USERNAME", payload: e.target.value })
          }
          type="text"
          name="username"
          placeholder="username ..."
        />
        <input
          value={state.input_password}
          onChange={(e) =>
            dispatch({ type: "SAVE_INPUT_PASSWORD", payload: e.target.value })
          }
          type="password"
          name="password"
          placeholder="password ..."
        />
        <div className="clickable">
          <Link href="/register">
            <a className="link" style={{ color: "var(--main-blue)" }}>
              Create an account.
            </a>
          </Link>
          <label htmlFor="checkbox">
            {" "}
            <input
              type="checkbox"
              id="checkbox"
              onChange={(e) =>
                dispatch({
                  type: "REMEMBER_ME_CHECK",
                  payload: e.target.checked,
                })
              }
              style={{ width: "20px", height: "40px" }}
            />
            Remember me
          </label>
        </div>
        <button onClick={handleSubmit} type="submit">
          login
        </button>
      </form>
    </div>
  );
}

export default LoginForm;
