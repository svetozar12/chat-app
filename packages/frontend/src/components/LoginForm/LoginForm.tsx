import { useSelector, useDispatch } from "react-redux";
import Link from "next/dist/client/link";
import { InitialState, InitialState3 } from "../../redux/state";
import styled from "@emotion/styled";
import { Alerts } from "../Alerts/Alerts";
import { css } from "@emotion/css";
export const Input = styled.input`
  width: 100%;
  height: 2rem;
  margin: 0.5rem 0;
  border: 1px solid var(--input-border-color);
  border-radius: 5px;
  transition: 0.3s;
  padding: 1.3rem 0.9rem;
`;

const CheckBox = styled.input`
  width: 20px;
  height: 40px;
`;

export const Form = styled.form`
  width: 40%;
  height: 50vh;
  padding: 3rem;
  border-radius: 5px;
  background: #fff;
  box-shadow: 0 2px 30px rgba(0, 0, 0, 0.1);
  @media (max-width: 1010px) {
    width: 70%;
  }
`;

export const Button = styled.button`
  margin-top: 1rem;
  border-radius: 5px;
  width: 60%;
  background-color: var(--button-blue);
  color: var(--main-white);
  border: 1px solid var(--input-border-color);
  padding: 1rem;
  cursor: pointer;
  transition: 0.2s;
  font-weight: bold;
  font-size: 1rem;
`;

const Clickable = styled.div`
  display: flex;
  align-items: center;
`;

const Link_anchor = styled.a`
  text-decoration: none;
  cursor: pointer;
  border: 1px solid transparent;
  margin: 1rem;
`;

export const Label_container = styled.div`
  dispay: flex;
  flex-direction: collumn;
  width: 60%;
`;

export const Form_header = styled.h1`
  padding: 1rem;
  height: auto;
  text-align: center;
  margin: 0;
  width: 40%;
  border-top-right-radius: 5px;
  border-top-left-radius: 5px;
  color: var(--main-white);
  background: var(--form-gray);
  @media (max-width: 1010px) {
    width: 70%;
  }
`;

function LoginForm({
  handleSubmit,
}: {
  handleSubmit: (e: React.MouseEvent<HTMLButtonElement>) => Promise<void>;
}) {
  const state = useSelector(
    (state: { authReducer: InitialState }) => state.authReducer,
  );

  const inputState = useSelector(
    (state: { saveInputReducer: InitialState3 }) => state.saveInputReducer,
  );
  const dispatch = useDispatch();

  return (
    <div style={{ height: "100vh", flexDirection: "column" }} className="flex">
      {(state.good || state.bad) && <Alerts />}
      <div style={{ width: "100%", height: "3rem" }}></div>
      <Form_header>Login</Form_header>
      <Form>
        <Label_container>
          <label>Username</label>
          <Input
            value={inputState.input_username}
            onChange={(e) =>
              dispatch({ type: "SAVE_INPUT_USERNAME", payload: e.target.value })
            }
            type="text"
            name="username"
            placeholder="username ..."
          />
        </Label_container>
        <Label_container>
          <label>Password</label>
          <Input
            value={inputState.input_password}
            onChange={(e) =>
              dispatch({ type: "SAVE_INPUT_PASSWORD", payload: e.target.value })
            }
            type="password"
            name="password"
            placeholder="password ..."
          />
        </Label_container>
        <Button onClick={handleSubmit} type="submit">
          Log In
        </Button>
        <Clickable>
          <Link href="/register">
            <Link_anchor
              className="link"
              style={{ color: "var(--button-blue)" }}
            >
              Sign up for chatApp .
            </Link_anchor>
          </Link>
          <CheckBox
            data-testid="checkbox"
            type="checkbox"
            id="checkbox"
            checked={state.remember_me}
            onChange={(e) =>
              dispatch({
                type: "REMEMBER_ME_CHECK",
                payload: e.target.checked,
              })
            }
          />
          <label style={{ marginLeft: ".25rem" }} htmlFor="checkbox">
            {" "}
            Remember me
          </label>
        </Clickable>
      </Form>
    </div>
  );
}

export default LoginForm;
