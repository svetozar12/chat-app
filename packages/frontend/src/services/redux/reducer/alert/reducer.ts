import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
