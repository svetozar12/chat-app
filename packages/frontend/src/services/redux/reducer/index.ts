import { AnyAction, combineReducers } from 'redux';
import authReducer from 'services/redux/reducer/auth/reducer';
import { ActionType } from '../types';
import { HYDRATE } from 'next-redux-wrapper';
import wsReducer from 'services/redux/reducer/websocket/reducer';
import messageReducer from 'services/redux/reducer/messages/reducer';

import { IAuth } from 'services/redux/reducer/auth/state';
import { IWebSocket } from 'services/redux/reducer/websocket/state';
import { IMessage } from 'services/redux/reducer/messages/state';
import IInputs from 'services/redux/reducer/inputs/state';
import inputReducer from 'services/redux/reducer/inputs/reducer';
import toggleReducer from 'services/redux/reducer/toggles/reducer';
import { IToggle } from 'services/redux/reducer/toggles/state';
import IInvite from 'services/redux/reducer/invites/state';
import inviteReducer from 'services/redux/reducer/invites/reducer';
import { IAlert } from 'services/redux/reducer/alert/state';
import alertReducer from 'services/redux/reducer/alert/reducer';

export interface STATE {
  auth: IAuth;
  ws: IWebSocket;
  messages: IMessage;
  inputs: IInputs;
  toggle: IToggle;
  invite: IInvite;
  alert: IAlert;
}

const combiReducers = combineReducers({
  auth: authReducer,
  ws: wsReducer,
  messages: messageReducer,
  inputs: inputReducer,
  toggle: toggleReducer,
  invite: inviteReducer,
  alert: alertReducer,
});

const combReducers = (state: any, action: AnyAction) => {
  if (action.type === HYDRATE)
    return {
      ...state,
      ...action.payload,
    };
  if (action.type === ActionType.SIGN_OUT) {
    return undefined;
  }
  return combiReducers(state, action);
};

export default combReducers;
