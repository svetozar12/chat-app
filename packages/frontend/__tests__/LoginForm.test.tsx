import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import LoginForm from "../src/components/LoginForm";
import renderer from "react-test-renderer";
import { AuthState } from "../src/redux/reducer/authReducer";
import { render, cleanup, RenderResult } from "@testing-library/react";
import { ReactTestRendererJSON } from "react-test-renderer";
import "@testing-library/jest-dom";

let component: ReactTestRendererJSON | ReactTestRendererJSON[] | null;
let container: RenderResult;
const Mock = jest.fn();
const handleSubmit = Mock.bind({});

beforeEach(() => {
  const mockStore = configureStore([]);
  const store = mockStore({
    authReducer: AuthState,
  });

  component = renderer
    .create(
      <Provider store={store}>
        <LoginForm handleSubmit={handleSubmit} />
      </Provider>,
    )
    .toJSON();

  container = render(
    <Provider store={store}>
      <LoginForm handleSubmit={handleSubmit} />
    </Provider>,
  );
});

afterEach(cleanup);

describe("Render connected React-redux page", () => {
  it("should create snapshot for <LoginForm/>", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render <LoginForm/>", () => {
    const test = container.getByText("Login");
    expect(test).toBeInTheDocument();
  });
  it("should call handleSubmit when button is clicked", () => {});
});
