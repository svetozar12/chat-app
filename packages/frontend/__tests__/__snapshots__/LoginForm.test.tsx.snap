// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <LoginForm/> 1`] = `
<div
  className="login-box"
>
  <form
    className="container"
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <h1
      className="alert"
      style={
        Object {
          "color": "red",
        }
      }
    >
      
    </h1>
    <h1>
      Login
    </h1>
    <input
      name="username"
      onChange={[Function]}
      placeholder="username ..."
      type="text"
      value=""
    />
    <input
      name="password"
      onChange={[Function]}
      placeholder="password ..."
      type="password"
      value=""
    />
    <div
      className="clickable"
    >
      <a
        className="link"
        href="/register"
        onClick={[Function]}
        onMouseEnter={[Function]}
        style={
          Object {
            "color": "var(--main-blue)",
          }
        }
      >
        Create an account.
      </a>
      <label
        htmlFor="checkbox"
      >
         
        <input
          id="checkbox"
          onChange={[Function]}
          style={
            Object {
              "height": "40px",
              "width": "20px",
            }
          }
          type="checkbox"
        />
        Remember me
      </label>
    </div>
    <button
      onClick={[MockFunction]}
      type="submit"
    >
      login
    </button>
  </form>
</div>
`;
